hmkit-android/src/main/java/com/highmobility/hmkit/GattServer.java

```diff
@@ -322,6 +356,75 @@ public void onCharacteristicReadRequest(final BluetoothDevice device,
                 offset,
                 offsetBytes);
 
+
+        /**
+         * Offset problem detection logic
+         *
+         * If detection is enabled
+         */
+        if (configuration.bleEnableOffsetProblemDetection()) {
```

Comment:
This is quite a bit of logic that is ran on every BLE write message. Should we move this to HMKit init instead? That way it is ran only once on app load